### ISSUE TITLE
Additional ProjectionManagement operations

### ISSUE THREAD

**Issue by patriknw** [akka/akka-projection#190]
In PR https://github.com/akka/akka-projection/pull/186 we discussed that `setOffset` is dangerous to use when the purpose is to skip failing retries.

Summary of some of the ideas:
* We can add `compareAndSetOffset` where the expected current offset is passed in as parameter and the operation will be ignored if current offset has changed. Checked after stopping the projection.
* Dedicated `skip` operator for the purpose to skip current failing retries. It would be possible to do this without restarting the projection. Would probably also need a current offset parameter to ensure that the projection is still stuck on the expected offset.

**Comment by patriknw** [comment]
Another idea is integrating the `StatusObserver` with `ProjectionManagement`, so that current error and restart status is available via the `ProjectionManagement`. We would then implement a built in `StatusObserver` that would keep track of things for each projection instance in memory, and the `ProjectionManagement` would query and gather the results via PubSub.

**Comment by patriknw** [comment]
See also #25

**Comment by jroper** [comment]
Assuming we're planning on keeping failure tracking and offset tracking separate, it's impossible to implement `skip` at this level of the API, since the current offset stored in the offset tracking table is the last successful offset, not the last failed offset. To skip an offset, you need to set the offset to be the last failed offset (which indicates that that offset was processed successfully, so processing will resume from after that offset, effectively skipping it).

I'm not sure about the value of `compareAndSet` - what's the use case? The use cases I see for setting the offset are:

* You want to replay the projection from some point in history. In which case, what the current offset is is irrelevant.
* Due to a bug in your projection code, you are blocked on an event that can never be processed successfully (without fixing the code), and as a quick/temporary fix to get production working again, you want to skip that event. Since you know the event can never be processed successfully, it's not possible that the current offset will change.

I can't think of any other use cases for setting the offset - an event that is temporarily failing to be processed that could start succeeding between when you check and when you set, you wouldn't want to skip, since the reason it's failing (eg database down) is likely to cause the next event to fail too.

**Comment by renatocaval** [comment]
I think you are right for the `compareAndSet`. It's true that if it's failing for an intermittent reason, you better wait or solve the source of the failure. 

I would argue that unfortunately that's not how people may use it.

I can pretty much see it happening. People will skip it and see that the next one is now failing and realise that the issue is elsewhere. Then a few moments later the system recovers and start to consume the events. And people will say: oh, but I skip one event, I need to rewind it now. Oh, I can't because my model is not idem-potent or because I will start to double publish in Kafka and the consumers will be impacted, etc. 

`compareAndSet` won't give any protection to the above cascading of bad actions though. I'm fine to not include a `compareAndSet`.  

However, I do see a problem in skipping arbitrary offsets, as least with the tooling we have now. We don't know which offset is failing, unless we capture it in logs or `StatusObserver`. For now, the way to do it is by saying: `skip currentOffset + 1`, where `currentOffset` is the last known successful offset. 

For Sequencial offsets, that's easy. For timebased ones, much harder. 

Also, skipping by offset is annoying. If event `Foo` is always failing, I need to keep watching and skip the offset each time a failure shows in the logs. 

My idea of changing the recovery strategy on the fly was to avoid it. Then, if I detect that something is misbehaving I can say: skip for the moment while I fix it and redeploy. 

We will need to identify exceptions that are happening inside the user handler. We can do it easily because we are already wrapping it in a try/catch. Those we want to treat differently. 

In other words, 'skip for the moment' means skip for the moment if an exceptions is thrown from the user handler, not all exceptions.
